samples/openthread/cli/CMakeLists.txt

```diff
@@ -6,12 +6,89 @@
+  nrf52840dk_nrf52840
+  nrf52833dk_nrf52833
 )
 
 find_package(Zephyr REQUIRED HINTS $ENV{ZEPHYR_BASE})
-
+include(${NRFXLIB_DIR}/common.cmake)
 project(openthread_cli)
-
 target_sources(app PRIVATE src/main.c)
+
+# Copy new OpenThread libraries, headers, text file with the configuration
+# to the nrfxlib openthread folder.
+function(ot_libs_substitute)
+  execute_process(
+    COMMAND           git rev-parse HEAD
+    WORKING_DIRECTORY ${openthread_SOURCE_DIR}
+    RESULT_VARIABLE   git_return
+    OUTPUT_VARIABLE   openthread_git_hash
+    )
+      
+  list(APPEND OT_SETTINGS "OpenThread_commit=${openthread_git_hash}\n\n")
+      
+  # Store all OT related variables
+  get_cmake_property(_variableNames VARIABLES)
+  list (SORT _variableNames)
+  foreach (_variableName ${_variableNames})
+    if("${_variableName}" MATCHES "^CONFIG_OPENTHREAD_.*|^OT_.*") 
+      list(APPEND OT_SETTINGS "${_variableName}=${${_variableName}}\n")
+    endif()
+  endforeach()
+      
+  FILE(WRITE 
+    "${PROJECT_BINARY_DIR}/openthread_lib_configuration.txt" 
+    ${OT_SETTINGS}
+    )
+  
+  set(OPENTHREAD_SRC_DIR "${PROJECT_BINARY_DIR}/modules/openthread/build/src")
+  set(OPENTHREAD_LIBS
+    "${OPENTHREAD_SRC_DIR}/core/libopenthread-ftd.a"
+    "${OPENTHREAD_SRC_DIR}/core/libopenthread-mtd.a"
+    "${OPENTHREAD_SRC_DIR}/core/libopenthread-radio.a"
+    "${OPENTHREAD_SRC_DIR}/cli/libopenthread-cli-ftd.a"
+    "${OPENTHREAD_SRC_DIR}/cli/libopenthread-cli-mtd.a"
+    "${OPENTHREAD_SRC_DIR}/ncp/libopenthread-ncp-ftd.a"
+    "${OPENTHREAD_SRC_DIR}/ncp/libopenthread-ncp-mtd.a"
+    "${OPENTHREAD_SRC_DIR}/ncp/libopenthread-rcp.a"
+  )
+
+  nrfxlib_calculate_lib_path(lib_path)
+
+  foreach(lib IN LISTS OPENTHREAD_LIBS)
+    add_custom_command(
+      TARGET app
+      COMMENT "Copying OpenThread library: ${lib}."
+      PRE_LINK
+      COMMAND ${CMAKE_COMMAND} -E copy "${lib}" 
+        "${NRFXLIB_DIR}/openthread/${lib_path}/v${OT_THREAD_VERSION}"
+      DEPENDS ${OPENTHREAD_LIBS}
+      )
+  endforeach()
+
+  add_custom_command(
+    TARGET app
+    COMMENT "Copying OpenThread headers."
+    PRE_LINK
+    COMMAND ${CMAKE_COMMAND} -E copy 
+      ${NRFXLIB_DIR}/openthread/include/openthread-config-generic.h 
+      ${PROJECT_BINARY_DIR}
+    COMMAND ${CMAKE_COMMAND} -E remove 
+      ${openthread_SOURCE_DIR}/include/openthread-config-android.h
+    COMMAND ${CMAKE_COMMAND} -E remove_directory -f 
+      ${NRFXLIB_DIR}/openthread/include
+    COMMAND ${CMAKE_COMMAND} -E copy_directory 
+      ${openthread_SOURCE_DIR}/include 
+      ${NRFXLIB_DIR}/openthread/include
+    COMMAND ${CMAKE_COMMAND} -E copy 
+      ${PROJECT_BINARY_DIR}/openthread-config-generic.h 
+      ${NRFXLIB_DIR}/openthread/include
+    COMMAND ${CMAKE_COMMAND} -E copy
+      ${openthread_SOURCE_DIR}/examples/platforms/openthread-system.h
+      ${NRFXLIB_DIR}/openthread/include/
+  )
+endfunction(ot_libs_substitute)
+
+if(CONFIG_OPENTHREAD_LIBRARIES_SUBSTITUTE)
+  ot_libs_substitute()
```

Comment:
Could we move the implementation of this function to the separate file and include it when the option is enabled? Just for the tideness.

Reply:
Maybe we should wait for https://github.com/nrfconnect/sdk-nrfxlib/pull/223 and then push a file with this function to the nrfxlib?

Reply:
@lmaciejonczyk, done.
Separated file common.cmake. Not sure about the location. So far it is placed in the sample dir.